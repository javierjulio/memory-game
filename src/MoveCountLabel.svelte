<script>
  let { count = 0 } = $props();
</script>

<div>
  {count} {(count === 1) ? "move" : "moves"}
</div>

<style>
  div {
    cursor: default;
    padding: .375rem .75rem;
  }
</style>
