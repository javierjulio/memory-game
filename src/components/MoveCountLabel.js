function MoveCountLabel({ count }) {
  const label = (count === 1) ? "move" : "moves"
  return (
    <div className="move-label">
      {count} {label}
    </div>
  )
}

export default MoveCountLabel
