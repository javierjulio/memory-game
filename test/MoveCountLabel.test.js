import { render } from '@testing-library/svelte'
import MoveCountLabel from '../src/MoveCountLabel.svelte'

describe('MoveCountLabel.svelte', () => {
  it('renders given move count', () => {
    const { container } = render(MoveCountLabel, { count: 0 });
    expect(container).toBeTruthy()
    expect(container.innerHTML).toContain('0 moves')
    expect(container.innerHTML).toMatchSnapshot()
  })

  it('renders label in singular form with count of 1', () => {
    const { container } = render(MoveCountLabel, { count: 1 });
    expect(container).toBeTruthy()
    expect(container.innerHTML).toContain('1 move')
    expect(container.innerHTML).toMatchSnapshot()
  })

  it('renders label in plural form with count of 2', () => {
    const { container } = render(MoveCountLabel, { count: 2 });
    expect(container).toBeTruthy()
    expect(container.innerHTML).toContain('2 moves')
    expect(container.innerHTML).toMatchSnapshot()
  })
})
