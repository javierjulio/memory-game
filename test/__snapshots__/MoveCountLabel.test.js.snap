// Vitest Snapshot v1

exports[`MoveCountLabel.svelte > renders given move count 1`] = `"<div><div class=\\"svelte-al6881\\">0 moves</div></div>"`;

exports[`MoveCountLabel.svelte > renders label in plural form with count of 2 1`] = `"<div><div class=\\"svelte-al6881\\">2 moves</div></div>"`;

exports[`MoveCountLabel.svelte > renders label in singular form with count of 1 1`] = `"<div><div class=\\"svelte-al6881\\">1 move</div></div>"`;
